client-common/src/key/private_key.rs

```diff
@@ -6,10 +6,81 @@ use zeroize::Zeroize;
 
 use crate::{ErrorKind, PublicKey, Result, ResultExt, SECP};
 
+/// types of private key
+#[derive(Clone)]
+pub enum PrivateKeyKind {
+    /// normal private key
+    Normal(PrivateKey),
+    // TODO: Add Hardware wallet as Privatekey
+}
+
+impl PrivateKeyTrait for PrivateKeyKind {
```

Comment:
If we just use the extensible `PrivateKeyKind` enum in logic code, no need for the trait `PrivateKeyTrait`, just normal method is enough?

Reply:
this would be useful for tests with some `TestOnlyKey{ connection_str: String}`

Reply:
PrivateKeyTrait 
it's alreay trait, choose something else,
refer to other coding paradigm